ete4/gtdb_taxonomy/gtdbquery.py

```diff
@@ -146,13 +146,29 @@ def _translate_merged(self, all_taxids):
 
     #     return taxid, spname, norm_score
 
-    def get_rank(self, taxids):
+    def _dirty_id_suffix(self, taxid):
+        pass
+
+    def _get_rank(self, taxids):
         """Return dictionary converting taxids to their GTDB taxonomy rank."""
         ids = ','.join('"%s"' % v for v in set(taxids) - {None, ''})
         result = self.db.execute('SELECT taxid, rank FROM species WHERE taxid IN (%s)' % ids)
         return {tax: spname for tax, spname in result.fetchall()}
+    
+    def get_rank(self, taxids):
+        taxid2rank = {}
+        name2ids = self._get_name_translator(taxids)
+        overlap_ids = name2ids.values()
+        taxids = [item for sublist in overlap_ids for item in sublist]
+        """Return dictionary converting taxids to their GTDB taxonomy rank."""
+        ids = ','.join('"%s"' % v for v in set(taxids) - {None, ''})
+        result = self.db.execute('SELECT taxid, rank FROM species WHERE taxid IN (%s)' % ids)
+        for tax, rank in result.fetchall():
+            taxid2rank[list(self._get_taxid_translator([tax]).values())[0]] = rank
+        
+        return taxid2rank
```

Comment:
I don't understand this change. Why do we have `get_rank()` and `_get_rank()`. What's their difference? Also, `get_rank()` would need a docstring, and we should avoid having similarly-named functions since that makes it hard to guess which one is appropriate to use.